dualscreeninfo/src/types.ts

```diff
@@ -1,18 +1,20 @@
-/**
- * Copyright (c) Microsoft Corporation.
- * Licensed under the MIT License.
- */
-
-export type DualScreenInfoEvent = 'spannedChange' | 'windowSizeChanged';
-
-export interface WindowRect {
-  width: number;
-  height: number;
-  x: number;
-  y: number;
-}
-
-export type SpannedChangeHandler = (
-  isSpanned: boolean,
-  windowRects: WindowRect
-) => void;
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+export type DualScreenInfoEvent = 'spannedChange' | 'windowSizeChanged';
```

Comment:
I realize this was already in the spec, but I have a timing question/concern.  Is it possible with this design for the UI to flicker due to changing their layout two separate times in response to these two separate events?  We really want the layout to change atomically as the window changes its orientation or size.

Reply:
not sure if there are any alternative to send events through the bridge

Reply:
The alternative would be to have a single event that fires when either the windowRects change, or the spanning changes.  But this might not be necessary.  I'd like someone to investigate if a flicker will result before changing the design.

---
In reply to: [382735389](https://github.com/microsoft/react-native-dualscreen/pull/2#discussion_r382735389) [](ancestors = 382735389)